SetupDataPkg/Library/ConfigKnobShimLib/ConfigKnobShimDxeLib/GoogleTest/ConfigKnobShimDxeLibGoogleTest.cpp

```diff
@@ -0,0 +1,117 @@
+/** @file
+  This is a sample to demonstrates the use of GoogleTest that supports host
+  execution environments.
+
+  Copyright (c) 2022, Microsoft Corporation. All rights reserved.<BR>
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#include <Library/GoogleTestLib.h>
+#include <GoogleTest/Library/MockUefiRuntimeServicesTableLib.h>
+extern "C" {
+#include <Uefi.h>
+#include <Library/BaseLib.h>
+#include <Library/DebugLib.h>
+
+// include the c file to be able to unit test static function
+#include "../../ConfigKnobShimLibCommon.c"
+}
+
+#define CONFIG_KNOB_GUID  {0x52d39693, 0x4f64, 0x4ee6, {0x81, 0xde, 0x45, 0x89, 0x37, 0x72, 0x78, 0x55}}
+
+using namespace testing;
+
+///////////////////////////////////////////////////////////////////////////////
+class GetConfigKnobOverrideFromVariableStorageTest : public Test {

```

Comment:
Assuming this works, I would remove the cmocka unit tests in this same PR, no need to keep both.